src/shared/modules/stream/streamDuck.ts

```diff
@@ -185,43 +189,99 @@ function ensureFrameLimit(state) {
   }
 }
 
-/** Inital state */
-export const initialState = {
+interface ConnectionData {
+  authEnabled: boolean
+  authenticatedMethod: string
+  db: null
+  host: string
+  id: string
+  name: string
+  password: string
+  type: string
+  username: string
+}
+
+interface FrameError {
+  cmd: string
+  code: string
+  message: string
+  type: string
+}
+
+// When more code is typed, frame should be a union type of different frames
+interface Frame {
+  autoCommit: boolean
+  cmd: string
+  connectionData: ConnectionData
+  error: FrameError
+  id: string
+  initialSlide: number
+  isRerun: boolean
+  parentId: string
+  query: string
+  result: string
+  requestId: string
+  statements: string[]
+  ts: number
+  type: string
+  useDb: string | null
+}
+
+interface FrameStack {
+  stack: Frame[]
+  isPinned: boolean
+}
+
+interface FramesState {
+  allIds: string[]
+  byId: { [key: string]: FrameStack }
+  recentView: null | FrameView
+  maxFrames: number
+}
+
+export const initialState: FramesState = {
   allIds: [],
   byId: {},
   recentView: null,
   maxFrames: 30
 }
 
-/**
- * Reducer
- */
-export default function reducer(state = initialState, action) {
+export default function reducer(
+  state = initialState,
+  action: StreamActions
+): FramesState {
   switch (action.type) {
     case APP_START:
       return { ...initialState, ...state }
     case ADD:
-      return addFrame(state, action.state)
+      return addFrame(state, (action as AddFrameAction).state)
```

Comment:
Use string literals instead of template strings for types, so you don't need to cast!